packages/base/src/lazy-tsp-client.ts

```diff
@@ -0,0 +1,55 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+
+import { TspClient } from 'tsp-typescript-client';
+
+/**
+ * Hack!
+ * When using actual classes as a type, TypeScript cares about private fields.
+ * In our case we only care about public APIs, not internal fields.
+ * This type only extracts public APIs out of `TspClient`.
+ */
+export type ITspClient = {
+    [K in keyof TspClient]: TspClient[K]
+};
```

Comment:
Having `TspClient` be a class with private properties makes it a bit annoying to mock.

Reply:
See [illustration](https://www.typescriptlang.org/play?#code/MYGwhgzhAEAqEAcDCICWBTAdgF2gbwFgAoaU6YAe0wmwCcBXYbC2gCmLM+gVtQDcw2dNABGkdAFVaIAFzQavTAHMOZAJT4AvqtKQAnpmDQAtumwALCgBNWauQAVaFY6gjoAPHwqorAPi3E2kTEAGb0htioVNDYYADW6PDIaFjYrKAYOHJJKJnYdtBePgHBRAD0ZdBIYJgA5LjokKggeiYUwHFwiLmpoujAYPRu0BQhMebCPPyCwiEYIFYAdMvElNS4xu1x2d0pONAAvPg60PqGJmaWNhqEJFykaxAUIOiLIBRKrLWbHehWtWoTkEgsQKtAAEL0XCobC1GCYCi4NAJaAAd2EEEs9AW0HQAA8EP1cISKAgXjEKOQwCAQNBUDhKRZXNxeAIhNA5ugFgBCVZUGjkPa4I6YdCorrJPJfMAygGrIWLMRuKQgADc0DB6FoTloxCAA).